File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/web/model/question/AnswerOption.java
package uk.co.claritysoftware.exam.slammr.webapp.web.model.question;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;

/**
 * Encapsulates the data for a possible answer of a question
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class AnswerOption {

    @NotBlank
    private String answer;

    private boolean correct;

    /**
     * @return a new AnswerOption instance with empty fields
     */
    public static AnswerOption generateEmptyAnswerOption() {
        return AnswerOption.builder().build();
    }
}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/web/model/question/AnswerOption.kt
package uk.co.claritysoftware.exam.slammr.webapp.web.model.question

import java.io.Serializable
import javax.validation.constraints.NotBlank

/**
 * Encapsulates the data for a possible answer of a question
 */
data class AnswerOption(@field:NotBlank
						var answer: String = "",

						var correct: Boolean = false
) : Serializable